veggieseasons/lib/screens/details.dart

```diff
@@ -6,68 +6,222 @@ import 'package:flutter/cupertino.dart';
+  // target.
+  Widget _buildVitaminText(int standardPercentage, Future<int> targetCalories) {
+    return FutureBuilder(
+      future: targetCalories,
+      builder: (context, snapshot) {
+        final target = snapshot?.data ?? 2000;
+        final percent = standardPercentage * 2000 ~/ target;
 
-  String get _firstChars {
-    return '${season.toString().substring(7, 8).toUpperCase()}'
-        '${season.toString().substring(8, 9)}';
+        return Text(
+          '$percent% DV',
+          textAlign: TextAlign.end,
+          style: Styles.detailsServingValueText,
+        );
+      },
+    );
   }
 
   @override
   Widget build(BuildContext context) {
+    final appState = ScopedModel.of<AppState>(context, rebuildOnChange: true);
+    final prefs = ScopedModel.of<Preferences>(context, rebuildOnChange: true);
+    final veggie = appState.getVeggie(id);
+
     return Padding(
-      padding: const EdgeInsets.all(4.0),
-      child: DecoratedBox(
-        decoration: BoxDecoration(
-          color: isHarvestTime
-              ? Styles.seasonColors[season]
-              : Styles.transparentColor,
-          borderRadius: BorderRadius.circular(25.0),
-          border: Styles.seasonBorder,
-        ),
-        child: SizedBox(
-          height: 50.0,
-          width: 50.0,
-          child: Center(
+      padding: const EdgeInsets.all(24.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          FutureBuilder(
+            future: prefs.preferredCategories,
+            builder: (context, snapshot) {
+              return Text(
+                veggie.categoryName.toUpperCase(),
+                style: (snapshot.hasData &&
+                        snapshot.data.contains(veggie.category))
+                    ? Styles.detailsPreferredCategoryText
+                    : Styles.detailsCategoryText,
+              );
+            },
+          ),
+          SizedBox(height: 8.0),
+          Text(
+            veggie.name,
+            style: Styles.detailsTitleText,
+          ),
+          SizedBox(height: 8.0),
+          Text(
+            veggie.shortDescription,
+            style: Styles.detailsShortDescriptionText,
+          ),
+          SizedBox(height: 24.0),
+          Padding(
+            padding: const EdgeInsets.only(
+              left: 9.0,
+              bottom: 4.0,
+            ),
             child: Text(
-              _firstChars,
-              style: isHarvestTime
-                  ? Styles.activeSeasonText
-                  : Styles.inactiveSeasonText,
+              'Serving info',
+              style: Styles.detailsServingHeaderText,
             ),
           ),
-        ),
+          Container(
+            decoration: BoxDecoration(
+              border: Border.all(color: Styles.servingInfoBorderColor),
+            ),
+            padding: const EdgeInsets.all(8.0),
+            child: Column(
```

Comment:
This build is kinda long. Maybe break this into another widget?

Reply:
Tables add so much rightward drift. :(

I'll break it out.